kubernetes/roles/deploy-kubevirt/tasks/main.yml

```diff
@@ -1,14 +1,67 @@
-- name: "List all manifests"
+---
+- include_role:
+    name: ansible.kubernetes-modules
+  when: wait_kubevirt_rollout
+
+- file:
+    path: "{{ manifests_path }}"
+    recurse: True
+    state: directory
+
+- name: git clone kubevirt master
+  git:
+    repo: "{{ repo }}"
+    dest: "{{ checkout_path }}"
+  register: checkout
+  delegate_to: localhost
+
+- name: Find All Templates
   find:
-    paths: "{{ go_path }}/{{ kubevirt_dir }}/manifests"
-    patterns: "*.yaml"
-  register: all_manifests_files
+    paths: "{{ manifests_path }}"
+    pattern: "^((?!demo).)*$"
+    use_regex: True
+    recurse: True
+  register: templates
+  delegate_to: localhost
+
+- name: Process Templates
+  template:
+    src: "{{ item.path }}"
+    dest: "{{ item.path | splitext | first }}"
+  with_items: "{{ templates.files }}"
+  register: template_files
+
+- command: "kubectl apply -f {{ item.dest | default(item.path) }} -n {{ namespace }}"
```

Comment:
Why is the default filter needed?

Reply:
@gbenhaim the result from the template module changes the variable which holds the path to the resulting templated file.  If there is no change `item.path` exists.  If there is a change `item.dest` exists.